stytch/shared/rbac_local.go

```diff
@@ -0,0 +1,42 @@
+package shared
+
+import (
+	"github.com/stytchauth/stytch-go/v11/stytch/b2b/rbac"
+	"github.com/stytchauth/stytch-go/v11/stytch/b2b/sessions"
+	"github.com/stytchauth/stytch-go/v11/stytch/stytcherror"
+)
+
+func PerformAuthorizationCheck(
+	policy rbac.Policy,
+	subjectRoles []string,
+	subjectOrgID string,
+	authorizationRequest *sessions.AuthorizationCheck,
+) error {
+	if subjectOrgID != authorizationRequest.OrganizationID {
+		return stytcherror.NewClientLibraryError("Subject organization ID does not match ID from request")
+	}
+
+	for _, role := range policy.Roles {
+		if doesListOfStringsContainTarget(subjectRoles, role.RoleID) {
+			for _, permission := range role.Permissions {
+				hasMatchingAction := doesListOfStringsContainTarget(permission.Actions, "*") ||
+					doesListOfStringsContainTarget(permission.Actions, authorizationRequest.Action)
+				hasMatchingResource := permission.ResourceID == authorizationRequest.ResourceId
+				if hasMatchingAction && hasMatchingResource {
+					return nil
+				}
+			}
+		}
+	}
+
+	return stytcherror.NewClientLibraryError("Member is not authorized")
+}
+
+func doesListOfStringsContainTarget(stringList []string, target string) bool {
+	for _, s := range stringList {
+		if target == s {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Go 1.21 just added `slices.Contains(list, target)`, but alas, we're still supporting 1.18 in this library. A good note to make for later, though.